### ISSUE TITLE
Reload feature in OMEdit

### ISSUE THREAD

**Issue by OpenModelica-TracImporter** [OpenModelica/OpenModelica#5664]

**Comment by bilderbuchi** [comment]
Yes, this would indeed be useful, especially if one uses another editor in parallel to do things OMEdit can't do (e.g. refactoring/renaming, etc.) or if using version control tools to undo damage wrought if OMEdits fallback formatter gets triggered.
An Unload-Reload cycle also destroys the navigation/open file state in the editor, which could be avoided with a "reload from disk" option.

**Comment by adrpo** [comment]
I have implemented Reload as Unload/Load which will not save the state because you don't know if you still have that file or model if you load the library from disk again. I will make a PR soon and you can test it and we can improve it if needed.

**Comment by bilderbuchi** [comment]
bummer about not saving the state. Is it difficult to "refresh" state and remove any models which don't exist anymore? Otherwise, all this does is save some clicks (which is still good!).

If you press "reload" on a model that has disappeared on disk, can you remove the model from display only if the load step fails? If yes, can that operation be recursed through a package?

**Comment by adrpo** [comment]
I think it should be possible to save most of the state. I will make the PR and then we can improve it further.

**Comment by adrpo** [comment]
I implemented this one as I had to reload a library a lot in the last days and unload, navigate to Welcome, select the file again took too much time. 
 
I agree that saving state in OMEdit need improvement. I will make a new ticket about it and see what we can do about it.

Myself I hate that:
- you cannot get back to the previous state after quitting OMEdit and the restarting it
- in the recent list you don't have classes, only files

**Comment by bilderbuchi** [comment]
@adrpo any chance to open a PR for the fix you implemented in  https://github.com/adrpo/OpenModelica/commit/202effd85b038fda0b1499afe5a32d8bdb0a85d0?

**Comment by adrpo** [comment]
@adeas31, what do you think about this? I guess we can make a quick reload feature and then look into the more complex state saving thing.

**Comment by adeas31** [comment]
Sure. Make a PR of your commit and I will test it.

**Comment by adrpo** [comment]
We have PR #9204. I will resurect that.

**Comment by bilderbuchi** [comment]
Any update on this, or a timeline? 
In the past few days I spent some time refactoring a package (and also fixing merge conflicts etc), and the frequent need for the unload-welcome screen-pick package-navigate to where I was before was annoying, and made me think of this issue again. :-)

**Comment by casella** [comment]
Will be done when @adeas31 is back from vacation. We just need to finalize PR #9204.

**Comment by casella** [comment]
@adeas31, @adrpo can we try to finalize this feature? It seems we are almost done with it. Thanks!

**Comment by casella** [comment]
@adeas31 will take care of that shortly.

**Comment by casella** [comment]
Great! Thanks @adeas31
